contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java

```diff
@@ -142,4 +145,203 @@ public boolean CryptUnprotectData(DATA_BLOB pDataIn, PointerByReference szDataDe
+			PointerByReference rgpbToBeSigned, IntByReference rgcbToBeSigned, Pointer pbSignedBlob,
+			IntByReference pcbSignedBlob);
+
+	/**
+	 * The CertGetCertificateChain function builds a certificate chain context
+	 * starting from an end certificate and going back, if possible, to a trusted
+	 * root certificate.
+	 * 
+	 * @param hChainEngine
+	 *            A handle of the chain engine (namespace and cache) to be used. If
+	 *            hChainEngine is NULL, the default chain engine, HCCE_CURRENT_USER,
+	 *            is used. This parameter can be set to HCCE_LOCAL_MACHINE.
+	 * @param pCertContext
+	 *            A pointer to the CERT_CONTEXT of the end certificate, the
+	 *            certificate for which a chain is being built. This certificate
+	 *            context will be the zero-index element in the first simple chain.
+	 * @param pTime
+	 *            A pointer to a FILETIME variable that indicates the time for which
+	 *            the chain is to be validated. Note that the time does not affect
+	 *            trust list, revocation, or root store checking. The current system
+	 *            time is used if NULL is passed to this parameter. Trust in a
+	 *            particular certificate being a trusted root is based on the
+	 *            current state of the root store and not the state of the root
+	 *            store at a time passed in by this parameter. For revocation, a
+	 *            certificate revocation list (CRL), itself, must be valid at the
+	 *            current time. The value of this parameter is used to determine
+	 *            whether a certificate listed in a CRL has been revoked.
+	 * @param hAdditionalStore
+	 *            A handle to any additional store to search for supporting
+	 *            certificates and certificate trust lists (CTLs). This parameter
+	 *            can be NULL if no additional store is to be searched.
+	 * @param pChainPara
+	 *            A pointer to a CERT_CHAIN_PARA structure that includes
+	 *            chain-building parameters.
+	 * @param dwFlags
+	 *            Flag values that indicate special processing. This parameter can
+	 *            be a combination of one or more of the following flags.
+	 * @param pvReserved
+	 *            This parameter is reserved and must be NULL.
+	 * @param ppChainContext
+	 *            The address of a pointer to the chain context created. When you
+	 *            have finished using the chain context, release the chain by
+	 *            calling the CertFreeCertificateChain function.
+	 * @return If the function succeeds, the function returns nonzero (TRUE). If the
+	 *         function fails, it returns zero (FALSE).
+	 */
+	boolean CertGetCertificateChain(Pointer hChainEngine, CERT_CONTEXT pCertContext, Pointer pTime,
+			Pointer hAdditionalStore, CERT_CHAIN_PARA pChainPara, int dwFlags, Pointer pvReserved,
+			CERT_CHAIN_CONTEXT ppChainContext);
```

Comment:
It would be good if a subclass of `HANDLE` would be introduces for `hChainEngine` and that the two predefined engines `HCCE_CURRENT_USER` and `HCCE_LOCAL_MACHINE` are bound.

`pTime` should be a `FILETIME`.

I would consider introducing `HCERTSTORE` as a subclass for `HANDLE` for the `hAdditionalStore` parameter.

It would be good if a unittests would be added, that verifies, that the output parameter `ppChainContext` is correctly filled.

Reply:
Updated these.